### ISSUE TITLE
Method scanner not ignoring lambda functions

### ISSUE THREAD

**Issue by kev626** [Fierioziy/ParticleNativeAPI#8]
The method scanner function “ tryInferMethodName” does not properly ignore lambda functions. This causes issues with private forks which contain lambda functions with matching parameters:

```
me.casperge.realisticseasons.particleapi.api.utils.ParticleException: Ambiguous names (a, lambda$send$14) for void method(Packet) in net.minecraft.server.network.PlayerConnection class.
	at me.casperge.realisticseasons.particleapi.core.asm.utils.RefUtils.tryInferMethodName(RefUtils.java:117) ~[RealisticSeasons-8.0.6 (1).jar:?]
```

Solving this is simple, simply ignore methods who’s name starts with “lambda$”. This method should not attempt to find lambda functions anyways as calling them in this way is unsafe.

**Comment by Fierioziy** [comment]
I'd rather not depend on method name itself, but on _bridge_ or _synthetic_ modifier presence.

Resolving bridge methods (despite being _synthetic_) is fine, but any other synthetic methods (that includes lambdas) will be filtered out.

I also would like to write few tests to cover this case, but I would have to bump target compile version to java 1.8 to do so and I hesitate to drop 1.7 just for that. Anyways, using 1.7 is unsafe nowadays so maybe I'll do it with this patch

**Comment by Fierioziy** [comment]
Fixed in 3.2.1

**Comment by kev626** [comment]
Perfect, thank you!
